pkg/controller/jobframework/reconciler.go

```diff
@@ -159,6 +159,11 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 				return ctrl.Result{}, err
 			}
 		}
+		if !object.GetDeletionTimestamp().IsZero() {
```

Comment:
mental note:
If `client.Get` can't find the object, the deletion timestamp is null, which `IsZero` knows how to handle.
/hold cancel